File: src/main/js/formulas/logic/FALSE.ts
import { SheetColumnRowKey } from "../../models/common/SheetColumnRowKey";
import { Complex } from "../../models/common/Types";
import { AbstractFormula } from "../AbstractFormula";
import { FormulaName } from "../FormulaName";

export class FALSE extends AbstractFormula {
  static SELF: FALSE = new FALSE();
  NAME = FormulaName.FALSE;

  internal(origin: SheetColumnRowKey, ...values: Array<Complex>) {
    AbstractFormula.checkLength(values.length, 0, this.NAME);
    return false;
  }
}

File: src/main/java/io/protobase/f7/formulas/logic/FALSE.java
package io.protobase.f7.formulas.logic;

import io.protobase.f7.formulas.AbstractFormula;
import io.protobase.f7.formulas.FormulaName;
import io.protobase.f7.models.GridColumnRowKey;

public class FALSE extends AbstractFormula {
  public static FormulaName NAME = FormulaName.FALSE;
  public static FALSE SELF = new FALSE();

  @Override
  public Object internal(GridColumnRowKey origin, Object... arguments) {
    checkLength(arguments.length, 0, NAME);
    return false;
  }
}
